primitives/src/merkle_tree/field_based_mht/parameters/bn382.rs

```diff
@@ -0,0 +1,112 @@
+use algebra::{
+    fields::bn_382::Fr as BN382Fr,
+    biginteger::BigInteger384,
+    field_new,
+};
+
+use crate::{
+    crh::poseidon::BN382FrPoseidonHash,
+    FieldBasedMerkleTreePrecomputedEmptyConstants,
+};
+
+// PoseidonHash("This represents an empty Merkle Root for a BN382PoseidonHash based Merkle Tree.")
+pub const BN382_PHANTOM_MERKLE_ROOT: BN382Fr =
+    field_new!(BN382Fr, BigInteger384([
+        4838402151703393731,
+        9090604893589195550,
+        1731006251955805765,
+        1348662098166494253,
+        8636612641322414294,
+        1080867906211844884
+     ]));
+
+pub const BN382_MHT_POSEIDON_PARAMETERS: FieldBasedMerkleTreePrecomputedEmptyConstants<'static, BN382FrPoseidonHash> =
+    FieldBasedMerkleTreePrecomputedEmptyConstants {
+        nodes: &[
```

Comment:
How are these constants determined?

Reply:
Re-generated using update BN382 Poseidon constants. They are simply the hash of empty nodes up to a certain height